apps/src/lib/node/ledger/tendermint_node.rs

```diff
@@ -74,8 +114,29 @@ pub async fn run(
         tokio::sync::oneshot::Sender<()>,
     >,
 ) -> Result<()> {
-    let home_dir_string = home_dir.to_string_lossy().to_string();
     let tendermint_path = from_env_or_default()?;
+
+    let version = get_version(&tendermint_path).await.map_err(|err| {
+        Error::Runtime(format!("Failed to check Tendermint version: {:?}", err))
+    })?;
+    let version_reqs = version_requirements();
+    if version_reqs.matches(&version) {
+        tracing::info!(
+            %tendermint_path,
+            %version,
+            %version_reqs,
+            "Running with supported Tendermint version",
+        );
+    } else {
+        tracing::warn!(
+            %tendermint_path,
+            %version,
+            %version_reqs,
+            "Running with a Tendermint version which is not supported"
+        );
```

Comment:
Here I actually think we should terminate the ledger by default if the `tendermint version` isn't compatible, to avoid accidentally clobbering our `tendermint` directory, but there needs to also be some easy way to override this for development and using custom Tendermint builds. This functionality could be a separate issue/PR to this one.

Reply:
you can override version requirements when the `TENDERMINT` env var is set, and the `dev` feature flag is on or something. or completely lift version requirements altogether when `dev` is on

Reply:
I'm in the middle of testing custom tendermint builds, and this is a bit inconvenient

Reply:
This PR won't cause the ledger to terminate if an unsupported version of Tendermint is being used, just log a warning. Have opened https://github.com/anoma/namada/issues/525 though to change this behaviour after this PR lands, but it is inconvenient if it's not easily overrideable so won't make that change without having an easy way to override it as well.

Reply:
@sug0 I've updated this log message, after #525 only users who explicitly override the defaults and use a custom/unsupported Tendermint build will be able to see this warning